### ISSUE TITLE
Computation of future quantity is broken

### ISSUE THREAD

**Issue by beuss** [axelor/axelor-business-suite#3456]
Spotted on a production isntance, reproduced with a fresh install of the dev branch.

- Create a purchase order for a product with a tracking number, quantity should be greater than the quantity per number (eg. EQPT-0005) for demo data;
- Note the future quantity of the product;
- Validate the purchase order and create a supplier delivery;
- Update the product page to reflect new future quantity of the product → the new quantity does not take the whole delivery into account.

Note that it works "sometimes", if I order 5 printers, the quantity is correct, for 20, only 5 get impacted.

A quick look seems to lead to com.axelor.apps.stock.service.StockLocationLineServiceImpl#findIncomingPlannedStockMoveLines/com.axelor.apps.stock.service.StockLocationLineServiceImpl#findOutgoingPlannedStockMoveLines which do no "see" the planned lines of the currently processed stock move since they're not yet recorded in DB

**Comment by beuss** [comment]
Forgot the "it work sometimes", it seems that it was just that I had a previous move of 5 pces that has been taken into account when planning a new one (since quantity matched it looks like it works)

Also it seems that this has consequences on purchase order line delivery status because you end up with [number of item per serial] delivered for the purchase order line instead of the whole quantity.

**Comment by ale-axelor** [comment]
Hello, and thank you for your help.
It happened because we recently changed how future quantity is computed, we now use a query, this way is more resilient to error.
I fixed this issue by saving the lines at the right status before updating locations. Now the query finds the lines that are being created.

The anomaly on the delivery status was another issue, it happened because of the tracking splitting the moves, losing the link to the purchase order line.
